Assets/Scripts/InGameMenuController.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using TMPro;
+
+///<summary> 
+/// This class is used to control the escape menu and all of it's buttons.
+///</summary> 
+
+public class InGameMenuController : MonoBehaviour
+{
+    [Tooltip("The GameObject of the Esc Menu")] [SerializeField] private GameObject escMenu;
+    [Tooltip("GameObject of TextMeshPro for Interactive Objects")] [SerializeField] private GameObject interactiveTextOBJ;
+
+
+    private TextMeshProUGUI interactiveText;
+    private PlayerController pc;
+    private bool internalPause;
+
+
+    void Start()
+    {
+        interactiveText = interactiveTextOBJ.GetComponent<TextMeshProUGUI>();
+        setInteractiveText("");
+        GameObject player = GameObject.Find("Player");
+        escMenu.SetActive(false);
+        pc = player.GetComponent<PlayerController>();
+        internalPause = pc.IsGamePaused();
+    }
+    private void Update()
+    {
+        if (pc.IsGamePaused() != internalPause) {
+            internalPause = pc.IsGamePaused();
+            if (pc.IsGamePaused() == true)
+            {
+                openMenu();
+            }
+            else {
+                escMenu.SetActive(false);
+            }
```

Comment:
I am curious about the function of internalPause. Isnn't pc.IsGamePaused() sufficient for verifying whether the game is paused?

Reply:
Oh, the statement `pc.IsGamePaused() !=  internalPause` is an XOR so thar the code only runs once when the two become out of sync